ds3-sdk/src/test/java/com/spectralogic/ds3client/metadata/PosixMetadataRestore_Test.java

```diff
@@ -0,0 +1,131 @@
+import com.spectralogic.ds3client.metadata.interfaces.MetadataRestoreListener;
+import com.spectralogic.ds3client.metadata.interfaces.MetadataStoreListener;
+import com.spectralogic.ds3client.networking.Headers;
+import com.spectralogic.ds3client.networking.Metadata;
+import com.spectralogic.ds3client.utils.MetaDataUtil;
+import com.spectralogic.ds3client.utils.MetadataKeyConstants;
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.PosixFileAttributes;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.List;
+import java.util.Set;
+
+import static com.spectralogic.ds3client.utils.MetadataKeyConstants.KEY_GID;
+import static com.spectralogic.ds3client.utils.MetadataKeyConstants.KEY_OS;
+import static com.spectralogic.ds3client.utils.MetadataKeyConstants.KEY_UID;
+import static java.nio.file.LinkOption.NOFOLLOW_LINKS;
+
+@RunWith(JUnit4.class)
+public class PosixMetadataRestore_Test {
+    private final File file  = new File(getClass().getClassLoader().getResource("LoremIpsum.txt").getFile());
+    private final String localOS = MetaDataUtil.getOS();
+
+
+
+
+    @Test
+    public  void restoreFileTimes_Test() throws Exception{
+        final BasicHeader basicHeader[] = new BasicHeader[3];
+        final BasicFileAttributes attr = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+        basicHeader[0] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_CREATION_TIME,String.valueOf(attr.creationTime().toMillis()));
+        basicHeader[1] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_ACCESS_TIME,String.valueOf(attr.lastAccessTime().toMillis()));
+        basicHeader[2] = new BasicHeader(MetadataKeyConstants.METADATA_PREFIX + MetadataKeyConstants.KEY_LAST_MODIFIED_TIME,String.valueOf(attr.lastModifiedTime().toMillis()));
+        final  Metadata metadata = genMetadata(basicHeader);
+        final PosixMetadataRestore posixMetadataRestore = new PosixMetadataRestore(metadata,file.getPath(),MetaDataUtil.getOS(),Mockito.mock(MetadataRestoreListener.class));
+        posixMetadataRestore.restoreFileTimes();
+        final BasicFileAttributes fileAttributes = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+        Assert.assertEquals(String.valueOf(fileAttributes.creationTime().toMillis()),String.valueOf(basicHeader[0].getValue()));
+        Assert.assertEquals(String.valueOf(fileAttributes.lastModifiedTime().toMillis()),String.valueOf(basicHeader[2].getValue()));
+    }
+
+
+      @Test
```

Comment:
This test fails on windows


java.lang.UnsupportedOperationException: View 'unix' not available

	at sun.nio.fs.AbstractFileSystemProvider.readAttributes(AbstractFileSystemProvider.java:91)
	at java.nio.file.Files.readAttributes(Files.java:1964)
	at java.nio.file.Files.getAttribute(Files.java:1869)
	at com.spectralogic.ds3client.metadata.PosixMetadataRestore_Test.restoreUserAndOwner(PosixMetadataRestore_Test.java:65)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)